### PR TITLE
Menu clickable component

### PR DESCRIPTION
**Pull request by pcln-james**
* Changed `onChange` to `onClick` being used for the click event for chips to fix a bug
* Added `clickableNode` prop to Menu to support an clickable component being used as the base component to open a menu

### CODE REVIEW

File: packages/menu/src/Menu/Menu.js [hunk 1]
```diff
@@ -15,6 +15,7 @@ function Menu({
   id,
   buttonText,
   buttonProps,
+  clickableNode,
```
Comment (pcln-james) [inline packages/menu/src/Menu/Menu.js:18]: anyone got a better name?

**Review by sdalonzo — COMMENTED** [review]
the changes themselves look fine, but this represents a breaking change. we should talk about this.

**Review by pcln-barry — APPROVED** [review]
LGTM based on conversation in Slack!

**Comment by pcln-james** [comment]
To summarize the conversation in Slack: This PR includes a backwards compatible bug fix for the Chip component which should have been its own PR and still can be if anybody cares enough, thoughts?

**Review by ryan-hunt-priceline — APPROVED** [review]

**Review by sdalonzo — APPROVED** [review]

**Pull request merged**